production/helm/loki/templates/provisioner/job-provisioner.yaml

```diff
@@ -92,21 +92,27 @@ spec:
             - /bin/bash
             - -exuc
             - |
+              # In case, the admin resources have already been created, the provisioner job
+              # does not write the token files to the bootstrap mount.
+              # Therefore, secrets are only created if the respective token files exist.
               {{- range .Values.enterprise.provisioner.additionalTenants }}
-              kubectl --namespace "{{ .secretNamespace }}" create secret generic "{{ include "enterprise-logs.provisionedSecretPrefix" $ }}-{{ .name }}" \
-                --from-literal=token-write="$(cat /bootstrap/token-write-{{ .name }})" \
-                --from-literal=token-read="$(cat /bootstrap/token-read-{{ .name }})"
+              ! test -s /bootstrap/token-write-{{ .name }} || \
```

Comment:
The negation in the logic took me a bit longer to process. Would `test -s /bootstrap/token-write-{{ .name }} &&` work instead? I personally find that logic a little easier to read.

Reply:
Agree, initially wrote it like that however, we want the bash command to return a successful status code, otherwise (if an error status code is returned) the installation of the provision job is not considered successful. 
Added a comment to justify this https://github.com/grafana/loki/pull/8467/commits/987b8125d3906b998d78b86b0ecf135329dabb4c 👍